core/src/main/scala/org/bitcoins/core/gcs/TempUtxoProvider.scala

```diff
@@ -0,0 +1,28 @@
+package org.bitcoins.core.gcs
+
+import org.bitcoins.core.protocol.transaction.{
+  TransactionOutPoint,
+  TransactionOutput
+}
+
+import scala.collection.mutable
+
+/**
+  * A temporary abstraction over anything that keeps track of the UTXO set
+  */
+abstract class TempUtxoProvider {
+
+  def getUtxo(
+      transactionOutPoint: TransactionOutPoint): Option[TransactionOutput]
+}
+
+class TempUtxoProviderImpl extends TempUtxoProvider {
+
+  val utxos: mutable.Map[TransactionOutPoint, TransactionOutput] =
```

Comment:
Maybe I'm missing something here, but it doesn't appear like  we're ever adding to these utxos?

Reply:
The current plan is to use this in tests which will construct this map by adding things, but no, we don't use this anywhere yet.